### PR TITLE
Changed service. Now it takes data and id from db

### PR DESCRIPTION
**Pull request by TopOleh**
Service work well. Dont forget made changes in your pages that you have already done. If you have time pls put data into db.
At the stage of the fist commit there is not working home page get images.

### CODE REVIEW

File: src/app/core/services/news.service.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@ import { Observable } from 'rxjs';
 })
 export class NewsService {
 
-  private _url: string = "assets/mockinfo/mocknews.json"
-  
+  private _url = ';assets/mockinfo/mocknews.json';
```
Comment (TsirohOleh) [inline src/app/core/services/news.service.ts]: ';a - ?

File: src/app/modules/home/home.component.ts [hunk 2]
```diff
@@ -13,7 +13,8 @@ export class HomeComponent implements OnInit {
   constructor(private _dishService: DishService) { }
 
   ngOnInit() {
-    this._dishService.getDishesSortedByRating(this.dishes);
+    this._dishService.getDishesSortedByRating()
+    .subscribe(data => this.dishes.push(...data));
```
Comment (TsirohOleh) [inline src/app/modules/home/home.component.ts]: Why not this.dishes = data;
and typings

File: src/app/modules/menu/menu.component.ts [hunk 3]
```diff
@@ -1,8 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { FirebaseService } from 'app/core/services/firebase.service';
-import { Dish } from 'app/core/interfaces/dish';
-import { map } from 'rxjs/operators'
-
+import { Subscription } from 'rxjs';
```
Comment (TsirohOleh) [inline src/app/modules/menu/menu.component.ts]: import { Subscription } from 'rxjs/Subscription';

File: src/app/modules/menu/menu.component.ts [hunk 4]
```diff
@@ -11,19 +9,11 @@ import { map } from 'rxjs/operators'
...
-    this.firebaseServise.getItems('salads')
-    .subscribe(items => this.menu.push(...items))
-    this.firebaseServise.getItems('hot meals').subscribe(items => this.menu.push(...items))
-    this.firebaseServise.getItems('soups').subscribe(items => this.menu.push(...items))
-    this.firebaseServise.getItems('pizza').subscribe(items => this.menu.push(...items))
-    this.firebaseServise.getItems('salads').subscribe(items => this.menu.push(...items))
-    this.firebaseServise.getItems('salads').subscribe(items => this.menu.push(...items))
-    console.log(this.firebaseServise.getDoc1().subscribe(items => this.docs.push(...items)));
+    this.firebaseServise.getItems()
+    .subscribe(data => this.menu.push(...data));
```
Comment (TsirohOleh) [inline src/app/modules/menu/menu.component.ts]: this.menu = data;
typings

File: src/app/shared/menu-item/menu-item.component.ts [hunk 5]
```diff
@@ -1,22 +1,26 @@
...
 @Component({
   selector: 'app-menu-item',
   templateUrl: './menu-item.component.html',
   styleUrls: ['./menu-item.component.scss']
 })
 export class MenuItemComponent implements OnInit {
-  @Input() dishes;
-  public isHovered: boolean;
+  @Input() image: string;
+  @Input() hover: boolean;
```
Comment (TsirohOleh) [inline src/app/shared/menu-item/menu-item.component.ts]: isHovered

File: src/app/shared/menu-item/menu-item.component.ts [hunk 6]
```diff
@@ -1,22 +1,26 @@
...
+  @Input() name: string;
 
-  mouseHovering(i) {
-    return this.dishes[i].isHovered = true;
-  }
 
-  mouseLeaving(i) {
-    return this.dishes[i].isHovered = false;
-  }
+mouseHovering(i) {
```
Comment (TsirohOleh) [inline src/app/shared/menu-item/menu-item.component.ts]: i ?

File: src/app/shared/menu-wrapper/menu-wrapper.component.ts [hunk 7]
```diff
@@ -1,12 +1,14 @@
 import { Component, OnInit, Input } from '@angular/core';
+import { DishHome } from 'app/core/interfaces/dish-home';
+
```
Comment (TsirohOleh) [inline src/app/shared/menu-wrapper/menu-wrapper.component.ts]: multiple blank lines

File: src/app/core/services/firebase.service.ts [hunk 8]
```diff
@@ -1,52 +1,29 @@
...
-          })
-          return actions;
-        }))
-        return this.items;
+  getItems(): Observable<Dish[]> {
+    console.log('sending the request...');
+    return this.afs.collection<Dish>('menu')
+    .snapshotChanges()
+      .pipe(
+        map(data => {
```
[the remark this answers is not in the document]
Comment (TopOleh) [reply src/app/core/services/firebase.service.ts]: Can't put there: Dish, throw an error

File: src/app/core/services/dish.service.ts [hunk 9]
```diff
@@ -1,24 +1,20 @@
 import { Injectable } from '@angular/core';
-import { HttpService } from './http.service';
 import { map } from 'rxjs/operators';
+import { FirebaseService } from './firebase.service';
+import { Dish } from '../interfaces/dish';
+import { Observable } from 'rxjs';
```
[the remark this answers is not in the document]
Comment (TopOleh) [reply src/app/core/services/dish.service.ts]: Cannot find Observable in rxjs/Observable

File: src/app/core/services/firebase.service.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (TsirohOleh) [reply src/app/core/services/firebase.service.ts]: map((data: Dish): Observable => {

File: src/app/shared/menu-item/menu-item.component.html [hunk 10]
```diff
@@ -1,37 +1,21 @@
...
+  <div class="container d-flex flex-column align-items-center position-absolute ingredients__list" *ngIf="!dish.isHovered">
+    <div class="row d-flex">
+      {{dish.info}}
+    </div>
+  </div>
+  <div class="card-body">
+    <h5 class="card-title text-center my-2">{{dish.name}}</h5>
+    <div class="row d-flex text-center flex-nowrap my-4">
+      <div class="col-6">
+        {{dish.price}}$
```
Comment (TsirohOleh) [inline src/app/shared/menu-item/menu-item.component.html]: You can use pipe here https://angular.io/api/common/CurrencyPipe

File: src/app/core/services/firebase.service.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (TopOleh) [reply src/app/core/services/firebase.service.ts]: Doesnt work... There should be another type from Firebase. If we can lets skip it now.

**Pull request merged**